acceptance/openstack/identity/v3/policies_test.go

```diff
@@ -0,0 +1,257 @@
+package v3
+
+import (
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/opentelekomcloud/gophertelekomcloud/acceptance/clients"
+	"github.com/opentelekomcloud/gophertelekomcloud/acceptance/tools"
+	"github.com/opentelekomcloud/gophertelekomcloud/openstack/identity/v3/agency"
+	"github.com/opentelekomcloud/gophertelekomcloud/openstack/identity/v3/policies"
+	th "github.com/opentelekomcloud/gophertelekomcloud/testhelper"
+)
+
+func TestPolicyLifecycle(t *testing.T) {
+	if os.Getenv("OS_TENANT_ADMIN") == "" {
+		t.Skip("Policy doesn't allow NewIdentityV3AdminClient() to be initialized.")
+	}
+
+	client, err := clients.NewIdentityV3AdminClient()
+	th.AssertNoErr(t, err)
+
+	aOpts := policies.CreateOpts{
+		DisplayName: tools.RandomString("policy-test-", 5),
+		Type:        "AX",
+		Description: tools.RandomString("Description-", 5),
+		Policy: policies.CreatePolicy{
+			Version: "1.1",
+			Statement: []policies.CreateStatement{
+				{Action: []string{
+					"obs:bucket:ListBucket",
+				},
+					Effect: "Allow",
+					Condition: map[string]map[string][]string{
+						"StringLikeAnyOfIfExists": {
+							"obs:prefix": []string{"ht"},
+						},
+						"StringNotEndWith": {
+							"g:UserName": []string{tools.RandomString("end_of_name-", 5)},
+						},
+					},
+					Resource: []string{"OBS:*:*:bucket:some-bucket", "OBS:*:*:bucket:test-bucket"},
+				},
+				{Action: []string{
+					"obs:bucket:ListBucket",
+				},
+					Effect: "Deny",
+					Condition: map[string]map[string][]string{
+						"StringEquals": {
+							"g:UserId": []string{tools.RandomString("dummy_id-", 5)},
+						},
+					},
+					Resource: []string{"OBS:*:*:bucket:bucket"},
+				},
+			},
+		},
+	}
+
+	newPolicy, err := policies.Create(client, aOpts).Extract()
+	th.AssertNoErr(t, err)
+
+	defer func() {
+		err = policies.Delete(client, newPolicy.ID).ExtractErr()
+		th.AssertNoErr(t, err)
+	}()
+
+	listOpts := policies.ListOpts{
+		DisplayName: newPolicy.DisplayName,
+	}
+
+	allPages, err := policies.List(client, listOpts)
+	th.AssertNoErr(t, err)
+
+	th.AssertEquals(t, newPolicy.Description, allPages.Roles[0].Description)
+	th.AssertEquals(t, newPolicy.Type, allPages.Roles[0].Type)
+	th.AssertEquals(t, newPolicy.DisplayName, allPages.Roles[0].DisplayName)
+	th.AssertEquals(t, newPolicy.Policy.Version, allPages.Roles[0].Policy.Version)
+	if !reflect.DeepEqual(newPolicy.Policy.Statement, allPages.Roles[0].Policy.Statement) {
+		t.Error("Statement parameters are different")
+	}
+
+	newName := tools.RandomString("policy-test-new-", 5)
+	newDescription := tools.RandomString("description-new-", 5)
+
+	bOpts := policies.CreateOpts{
```

Comment:
aOpts and bOpts?

Reply:
First and second opts - one used for creation, another for resource update.
Should I rename them?

Reply:
use more descriptive names is better

Reply:
fixed